docs/core/integration-quality-scale/rules/iqs020.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "IQS020 - Reauthentication flow"
+---
+
+IQS020 is as follows:
+
+Reauthentication flow
+
+## Reasoning
+
+It can happen that users change their password of a device or service and forget that their device or account is still linked to Home Assistant.
+To avoid that the user has to remove the configuration entry and re-add it, we start a reauthentication flow.
+During this flow, the user can provide the new credentials to use from now on.
+
+This is a very user-friendly way to let the user know that they need to take action and update their credentials.
+
+## Example implementation
+
+In the example below, we show an authentication flow that allows the user to reauthenticate with a new API token.
+When we receive the new token, we check if we can connect to the service to avoid the user from entering an invalid token.
+If the connection is successful, we update the configuration entry with the new token.
+
+`config_flow.py`:
+```python
+class MyConfigFlow(ConfigFlow, domain=DOMAIN):
+    """My config flow."""
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauthentication upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm reauthentication dialog."""
+        errors: dict[str, str] = {}
+        if user_input:
+            client = MyClient(user_input[CONF_HOST], user_input[CONF_API_TOKEN])
+            try:
+                user_id = await client.check_connection(
+                    user_input[CONF_API_TOKEN],
```

Comment:
I don't think we need to pass the token again since the client already got the token when we created the client. That would be a weird API.